core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -920,9 +920,9 @@ private[spark] object Utils extends Logging {
       // user has access to them.
       randomizeInPlace(getYarnLocalDirs(conf).split(","))
     } else if (conf.getenv("SPARK_EXECUTOR_DIRS") != null) {
-      conf.getenv("SPARK_EXECUTOR_DIRS").split(File.pathSeparator)
+      randomizeInPlace(conf.getenv("SPARK_EXECUTOR_DIRS").split(File.pathSeparator))
```

Comment:
Hm, quick question though. `getConfiguredLocalDirs` gets invoked multiple times, and won't it return a different directory every time when it's called?

This `spark.local.dir` can be used to pass `spark.files` around, e.g., `SparkFiles.get`. Wonder if that would be affected.

If this is a K8S specific issue, I think it's best to localize the fix into K8S.